File: MODEL/angolo.cpp
#include "angolo.h"

#include<iostream>


Angolo::Angolo(double val):gradi(val){}

const Angolo Angolo::angoloRetto=Angolo(90);

double Angolo::getAngolo() const{
    return gradi;
}

bool Angolo::operator >(const Angolo a)const{
    return gradi > a.gradi ;
}

bool Angolo::operator <(const Angolo a)const{
    return gradi < a.gradi ;
}

bool Angolo::operator ==(const Angolo a)const{
    return gradi == a.gradi ;
}

bool Angolo::operator !=(const Angolo a)const{
    return gradi != a.gradi ;
}

bool Angolo::angPiatto(const Angolo a)const{
    double min = 179.9, max = 180.02, val=0;    //per pb. macchina
    val = a.gradi+gradi;
    return ( min < val && val < max );
}

Angolo Angolo::operator+(const Angolo& ang)const{
    return Angolo( gradi + ang.gradi);
}

Angolo& Angolo::operator=(const Angolo& ang){
    this->gradi=ang.gradi;
    return *this;
}

double Angolo::coseno()const{
    double coseno = cos( getAngolo() * PI/180);
    if(-0.01<coseno && coseno<0.01)
        coseno=0;
    return coseno;
}

double Angolo::seno()const{
    double seno = sin( getAngolo() * PI/180);
    if(-0.01<seno && seno<0.01)   //per pb. macchina
        seno=0;
    return seno;
}

File: KalkJava/src/Angolo.java
import java.lang.Math;

public class Angolo {

	private Double gradi;
	final static Angolo angoloRetto = new Angolo(90d);    //costante

	public Angolo(Angolo a) {
		gradi = a.gradi;
	}

	public Angolo(Double val) {
		gradi = val;
	}

	public Angolo() {
		this(0.0);
	}

	public Double getAngolo() {
		return gradi;
	}

	public boolean maggiore(Angolo a) {    //verificare
		if (gradi > a.gradi) return true;
		else return false;
	}

	public boolean minore(Angolo a) {
		if (gradi < a.gradi) return true;
		else return false;
	}

	public boolean equals(Angolo a) {    //ridefinizioine di ==		//il != teniamo !equals
		if (gradi == a.gradi) return true;
		else return false;
	}

	public boolean angPiatto(Angolo a) {

		Double min = 179.9;
		Double max = 180.02;
		Double val = 0d;    //per pb. macchina

		val = a.gradi + gradi;
		return (min < val && val < max);
	}

	public Angolo plus(Angolo ang) {        //operatore di somma degi angoli
		return new Angolo(gradi + ang.gradi);
	}

	public Double coseno() {
		Double coseno = Math.cos(getAngolo() * Math.PI / 180);
		if (-0.01 < coseno && coseno < 0.01) coseno = 0d;
		return coseno;
	}

	public Double seno() {
		Double seno = Math.sin(getAngolo() * Math.PI / 180);
		if (-0.01 < seno && seno < 0.01)   //per pb. macchina
			seno = 0d;
		return seno;
	}

	public static void main(String[] args) {

	}
}
